src/lib/Preview.js

```diff
@@ -782,6 +848,23 @@ class Preview extends EventEmitter {
     }
 
     /**
+     * Enrich file entries with offline links.
+     *
+     * @private
+     * @return {void}
+     */
+    enrichOfflineLinks() {
```

Comment:
Thank you guys for the fast and deliberate response. 

@tonyjin I also agree that this solution requires rethinking and you approach sound good. 
After better understanding about `addOriginalRepresentation` method, it made me think that we can enrich file info with offline file links on our side (property `authenticated_download_url` also, which base for creating original representation). So we can download all content files on our side after that put there to the storage and enrich `file` with their offline links. 

@priyajeet Seems like your solution requires some refactoring which can affect core part of box previewer so it will be overtime for me, but I like some solutions.  

So, guys, I've updated pull request with the latest rethinking, what you think?